zulipterminal/helper.py

```diff
@@ -86,9 +86,16 @@ def set_count(id_list: List[int], controller: Any, new_count: int) -> None:
         if msg['type'] == 'stream':
             key = (messages[id]['stream_id'], msg['subject'])
             unreads = unread_counts['unread_topics']
-        else:
+        # self-pm has only one display_recipient
+        # 1-1 pms have 2 display_recipient
+        elif len(msg['display_recipient']) <= 2:
             key = messages[id]['sender_id']
             unreads = unread_counts['unread_pms']  # type: ignore
+        else:  # If it's a group pm
```

Comment:
@amanagr I appreciate that you proposed this commit as a quick bugfix, but I would suggest we skip this second conditional, perhaps with a FIXME marker, particularly if you intend to work on this in a more consistent way?

One reason I think we should hold off is that particularly with the `type: ignore`s, we are skipping being aware that we are adding a frozenset key to the PMs, which shouldn't be working with that type, at least based on the Index. I would actually suggest we have a separate entry for 'huddles' in the index to make that more obvious?

Reply:
I have separated out the group pm additions into a separate commit and stored unread count of group pms in `unread_huddles`. I think we can merge it for now, so that atleast we properly reduce the unread count of correct pm.